VirtoCommerce.PricingModule.Data/Handlers/DeletePricesProductChangedEvent.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Catalog.Events;
+using VirtoCommerce.Domain.Catalog.Model;
+using VirtoCommerce.Domain.Common.Events;
+using VirtoCommerce.Domain.Pricing.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+
+namespace VirtoCommerce.PricingModule.Data.Handlers
+{
+    public class DeletePricesProductChangedEvent : IEventHandler<ProductChangedEvent>
+    {
+
+        private readonly IPricingService _pricingService;
+
+        public DeletePricesProductChangedEvent(IPricingService pricingService)
+        {
+            _pricingService = pricingService;
+        }
+
+        public virtual Task Handle(ProductChangedEvent message)
+        {
+
+            foreach (var changedEntry in message.ChangedEntries.ToList())
+            {
+                DeletePrices(changedEntry);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private void DeletePrices(GenericChangedEntry<CatalogProduct> changedEntry)
+        {
+            if (changedEntry.EntryState == EntryState.Deleted)
+            {
+                var product = changedEntry.OldEntry;
+
+                _pricingService.DeletePrices(product.Prices.Select(p => p.Id).ToArray());
```

Comment:
You can't use `Product.Prices` property for pricing deletion, this property doesn't contain prices and used for other purposes. 
Use this approach to get all product prices for future deletion https://github.com/VirtoCommerce/vc-module-pricing/blob/master/VirtoCommerce.PricingModule.Web/Controllers/Api/PricingModuleController.cs#L184